src/app/rpmostree-builtin-upgrade.cxx

```diff
@@ -129,6 +164,53 @@ rpmostree_builtin_upgrade (int             argc,
         g_print ("note: automatic updates (%s) are enabled\n", policy);
     }
 
+  if (!opt_bypass_driver)
+    {
+      g_autofree char *update_driver_sd_unit = NULL;
+      g_autofree char *update_driver_name = NULL;
+      if (!get_driver_info (&update_driver_name, &update_driver_sd_unit, error))
+        return FALSE;
+
+      /* Notify user that an updates driver is registered and upgrades should be
+       * done through the driver. */
+      if (update_driver_sd_unit && update_driver_name)
+        {
+          g_autoptr(GString) error_msg = g_string_new(NULL);
+          g_string_printf (error_msg, "Updates are driven by %s (%s)\n",
+                           update_driver_name, update_driver_sd_unit);
+          /* only try to get unit's `Documentation` if we're on the system bus */
+          if (bus_type == G_BUS_TYPE_SYSTEM)
+            {
+              g_autoptr(GVariant) update_driver_docs_array = NULL;
+              g_autoptr(GError) local_error = NULL;
+              GDBusConnection *connection =
+                g_dbus_proxy_get_connection (G_DBUS_PROXY (sysroot_proxy));
+              if (!get_update_driver_doc (connection, update_driver_sd_unit, &update_driver_docs_array,
+                                          cancellable, &local_error))
+                {
+                  g_printerr ("%s", local_error->message);
```

Comment:
We should probably distinguish between failing to query the systemd unit and the unit not having any `Documentation` keywords. Seems like in the latter case we should just print: "See %s's documentation."

Reply:
To confirm: does `get_update_driver_doc` return `TRUE` even if there's no `Documentation` in the unit?

Reply:
My understanding was that there's always `Documentation` in Unit Objects, but it could potentially be an empty array. So, in `get_update_driver_doc`, if there's no `Documentation` in the cache, an error is thrown.